pkg/datapath/iptables/iptables.go

```diff
@@ -214,28 +215,62 @@ var ciliumChains = []customChain{
 	},
 }
 
+// IptablesManager manages the iptables-related configuration for Cilium.
+type IptablesManager struct {
+	ip6tables bool
+}
+
+// Init initializes the iptables manager and checks for iptables kernel modules
+// availability.
+func (m *IptablesManager) Init() {
+	modulesManager := &modules.ModulesManager{}
+	ip6tables := true
+	if err := modulesManager.Init(); err != nil {
+		log.WithError(err).Fatal(
+			"Unable to get information about kernel modules")
+	}
+	if err := modulesManager.FindOrLoadModules(
+		"ip_tables", "iptable_nat", "iptable_mangle", "iptable_raw",
+		"iptable_filter"); err != nil {
+		log.WithError(err).Fatal("iptables modules could not be initialized")
+	}
+	if err := modulesManager.FindOrLoadModules(
+		"ip6_tables", "ip6table_mangle", "ip6table_raw"); err != nil {
+		if option.Config.EnableIPv6 {
+			log.WithError(err).Fatal(
+				"IPv6 is enabled and ip6tables modules could not be initialized")
+		}
+		log.WithError(err).Debug(
+			"ip6tables kernel modules could not be loaded, so IPv6 cannot be used")
+		ip6tables = false
+	}
+	m.ip6tables = ip6tables
```

Comment:
If ipv6 is disabled in the config, then we will still try to manage/delete those ipv6 rules so long as the modules are loaded correctly. Is this right, or am I missing something?

Reply:
Yes, this is right. The `ip6tables` field is going to be `true` every time the ip6tables modules are available. It's only false when those modules are not available to load.

Reply:
This seems a bit counter-intuitive, I'm wondering why would we manage ip6tables rules if ipv6 is disabled in Cilium?

Reply:
Because someone might decide to change the configuration and restart Cilium. For example, first have Cilium with ipv6 enabled and then change it to disabled. In such a case I think we should remove/cleanup ip6tables rules.